nox/_options.py

```diff
@@ -97,11 +97,30 @@ def _color_finalizer(value, args):
     return sys.stdin.isatty()
 
 
-def _posargs_finalizer(value, unused_args):
-    """Removes any leading "--"s in the posargs array."""
-    if value and value[0] == "--":
-        value.pop(0)
-    return value
+def _posargs_finalizer(value, args):
+    """Removes the leading "--"s in the posargs array (if any) and asserts that
+    remaining arguments came after a "--".
+    """
+    posargs = value
+    if not posargs:
+        return []
+
+    if "--" not in posargs:
+        unexpected_posargs = posargs
+        raise _option_set.ArgumentError(
+            None, "Unknown argument(s) '{}'.".format(" ".join(unexpected_posargs))
+        )
+
+    dash_index = posargs.index("--")
+    if dash_index != 0:
+        unexpected_posargs = posargs[0:dash_index]
+        raise _option_set.ArgumentError(
+            None, "Unknown argument(s) '{}'.".format(" ".join(unexpected_posargs))
+        )
+
+    # fmt: off
```

Comment:
Interesting, why did we have to turn black off for this?

Reply:
flake8 competes with black here. If I let black run, this is what flake8 does:
```
nox > flake8 nox tests
nox/_options.py:120:34: E203 whitespace before ':'
nox > Command flake8 nox tests failed with exit code 1
nox > Session lint failed.
```

I could add E203 to the ignore list in flake8 if you want (that would be my personal preference).